File: Data Structures & Algorithms/validate-parentheses/submission-5.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c in '{([':
                stack.append(c)
            else:
                try:
                    if c == '}' and stack.pop() != '{':
                        return False
                    elif c == ']' and stack.pop() != '[':
                        return False
                    elif c == ')' and stack.pop() != '(':
                        return False
                except:
                    return False;
        if len(stack) == 0:
            return True
        return False

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.cpp
class Solution {
public:
    int evalRPN(std::vector<std::string>& tokens) {
        std::stack<int> s;

        for (const auto& num : tokens) {
            if (num == "+" || num == "-" || num == "*" || num == "/") {
                int sec = s.top(); s.pop();
                int fir = s.top(); s.pop();

                if (num == "+") s.push(fir + sec);
                else if (num == "-") s.push(fir - sec);
                else if (num == "*") s.push(fir * sec);
                else if (num == "/") s.push(fir / sec);
            } else {
                s.push(std::stoi(num));
            }
        }
        
        int result = s.top();
        s.pop();
        return result;
    }
};